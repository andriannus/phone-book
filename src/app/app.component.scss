@import "@amar-ui-web/core";

$aligns: center, justify, left, right;

@each $align in $aligns {
  .TextAlign-#{$align} {
    text-align: $align;
  }
}
