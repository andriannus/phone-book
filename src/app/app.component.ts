import { defineComponent, h, VNode } from "vue";
import { RouterView } from "vue-router";

import "./app.component.scss";

export default defineComponent({
  name: "App",

  setup() {
    return (): VNode => h(RouterView);
  },
});
