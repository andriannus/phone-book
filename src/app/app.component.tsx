import { defineComponent } from "vue";
import { RouterView } from "vue-router";

import "./app.component.scss";

export default defineComponent({
  name: "App",

  setup() {
    return (): JSX.Element => <RouterView />;
  },
});
