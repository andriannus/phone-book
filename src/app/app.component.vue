<template>
  <router-view></router-view>
</template>

<script src="./app.component.js"></script>
<style lang="scss" src="./app.component.scss"></style>
