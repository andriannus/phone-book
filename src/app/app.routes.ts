import { createRouter, createWebHistory, RouteRecordRaw } from "vue-router";

import landingRoutes from "@/app/landing/landing.routes";
import notFoundRoutes from "@/app/not-found/not-found.routes";

const routes: RouteRecordRaw[] = [...landingRoutes, ...notFoundRoutes];

const router = createRouter({
  history: createWebHistory(process.env.BASE_URL),
  routes,
});

export default router;
