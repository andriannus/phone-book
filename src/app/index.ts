export { default as App } from "./app.component";
export { default as router } from "./app.routes";
