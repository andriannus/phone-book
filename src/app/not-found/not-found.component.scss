.NotFound {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: center;
  min-height: 100vh;
}