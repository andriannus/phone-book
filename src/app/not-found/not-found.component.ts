import { defineComponent, h, VNode } from "vue";
import { RouterLink } from "vue-router";
import { useTitle } from "@vueuse/core";

export default defineComponent({
  name: "NotFound",

  setup() {
    useTitle("Page Not Found | Phone Book");

    return (): VNode =>
      h(
        "div",
        {
          class: "NotFound",
        },
        [
          h(
            "p",
            {
              class: "FontSize-2xlarge MarginBottom",
            },
            "Page Not Found",
          ),

          h(
            RouterLink,
            {
              id: "BtnBackToLanding",
              class: "Button Button--primary",
              to: "/landing",
            },
            () => "Take me away",
          ),
        ],
      );
  },
});
