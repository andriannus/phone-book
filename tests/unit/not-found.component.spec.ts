import { shallowMount, VueWrapper } from "@vue/test-utils";

import { NotFound } from "@/app/not-found";

describe("not-found.component.vue", () => {
  let wrapper: VueWrapper<any>;

  beforeEach(() => {
    wrapper = shallowMount(NotFound);
  });

  it("should create", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
