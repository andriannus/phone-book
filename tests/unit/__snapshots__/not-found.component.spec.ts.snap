// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not-found.component.vue should create 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "__emits": Object {},
      "__props": Array [
        Object {},
        Array [],
      ],
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      <div
        class="NotFound"
      >
        <p
          class="FontSize-2xlarge MarginBottom"
        >
          Page Not Found
        </p>
        <router-link
          class="Button Button--primary"
          id="BtnBackToLanding"
          to="/landing"
        >
           Take me away 
        </router-link>
      </div>
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
          "props": Object {},
          "render": [Function],
        },
        "transition-group": Object {
          "name": "transition-group",
          "props": Object {},
          "render": [Function],
        },
      },
      "config": Object {
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isCustomElement": [Function],
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "mixins": Array [
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
        Object {
          "__emits": null,
          "__props": Array [],
          "beforeCreate": [Function],
        },
      ],
      "provides": Object {},
      "reload": [Function],
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isCustomElement": [Function],
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.0.4",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "__emitted": Object {},
  },
  "rootVM": Object {
    "__emitted": Object {},
  },
}
`;
